apps/src/blockly/addons/cdoXml.js

```diff
@@ -59,6 +59,12 @@ export default function initializeBlocklyXml(blocklyWrapper) {
       });
     });
 
+    // Below this point, we will be repositioning blocks on a rendered workspace.
```

Comment:
Would it be possible to add `reposition` as an optional third argument to this function, rather than relying on this early return? Seems like something we could accidentally break.

Reply:
Looking at this again... I think that the code below this point doesn't really need to be contained in `domToBlockSpace`, since at this point, the blocks are on the workspace. Instead, what if we moved all the repositioning logic to a separate function, and then updated our calls like this:

_Before:_
`Blockly.Xml.domToBlockSpace(workspace, xml);`

_After:_
```
    const xmlBlocks = Blockly.Xml.domToBlockSpace(workspace, xml);
    if (workspace.rendered) {
      positionBlocks(xmlBlocks, workspace);
    }
```

Reply:
Even better!